code-collab-app/src/editorpage.js

```diff
@@ -6,24 +6,20 @@ import logo from './codeCollabLogo.png';
 import './App.css';
 import IconButton from '@material-ui/core/IconButton';
 import AddIcon from '@material-ui/icons/Add';
+import { inject, observer } from 'mobx-react';
 
 class Editorpage extends React.Component {
   constructor(props){
     super(props);
 
     this.state = {
-      cellCount: 0,
-      cellText: (this.props.location.data) ? this.props.location.data : [],
-      cellLocked: [],
-      cellCurrentlyEditing: null,
+      cellText: this.props.peer_data.doc_data,
+      cellLocked: this.props.peer_data.cell_locked,
```

Comment:
@paulchenpmc @CarolineZhou Turning a `prop` into a `state` works okay as long as we never use `setState`, but just a style suggestion. Since we already have these variables in `props`, we don't really need these variables to be turned into `state` variables.

In that case we could:
- Delete the whole constructor method
- Replace all references to `this.state.cellText` with `this.props.peer_data.doc_data`
- Replace all references to `this.state.cellLocked` with `this.props.peer_data.cell_locked`

Reply:
Sure, I will change that in the next set of commits. This PR already have some conflicts so I want to finish merging this first. After that I will make those changes along with the next functionality I am working on.